### ISSUE TITLE
New eject bare workflow unable to load bundled assets on Release configuration and crashes

### ISSUE THREAD

**Issue by igaln** [expo/expo-cli#1859]
## Expected Behavior

After Expo Eject (new bare workflow),
expo publish
react-native run-ios --configuration Release 
application should run on device/simulator

## Observed Behavior

react-native run-ios --configuration Release 
crash with

```
2020-04-10 18:05:17.460983+0300 collect[8448:2356523] *** Terminating app due to uncaught exception ‘NSInvalidArgumentException’, reason: '* -[NSFileManager copyItemAtPath:toPath:options:error:]: source path is nil’**
```
Checked the urls, and see 
/var/mobile/Containers/Data/Application/C72454CA-B3C2-4E13-9DA1-EA4B70E8FA4C/Library/Application Support/.expo-internal/43b6c00fa250b3b2925695018d6fb84e0799a288213499970aa9dc2ab615aadf.ttf

The fonts are not being published to .expo-internal/ 

## Environment

Please run `expo diagnostics` in the project folder and copy the output here:

Also specify:

- Phone/emulator/simulator platform, model and version:

  Expo CLI 3.17.21 environment info:
    System:
      OS: macOS 10.15.4
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 12.16.2 - ~/.nvm/versions/node/v12.16.2/bin/node
      Yarn: 1.22.4 - /usr/local/bin/yarn
      npm: 6.14.4 - ~/.nvm/versions/node/v12.16.2/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.5 AI-191.8026.42.35.5791312
      Xcode: 11.4/11E146 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^37.0.0 => 37.0.6
      react: ~16.9.0 => 16.9.0
      react-native: ~0.61.5 => 0.61.5
    npmGlobalPackages:
      expo-cli: 3.17.21

## Reproducible Demo

Create Expo Project
Add Fonts
```
  async UNSAFE_componentWillMount() {
    await Font.loadAsync({
      //   Fonts
      // Maax
      Maax: require('../src/assets/font/Maax/Maax.otf'),
      Maax_Bold: require('../src/assets/font/Maax/Maax-Bold.otf'),
      Maax_BoldItalic: require('../src/assets/font/Maax/Maax-BoldItalic.otf'),
```

Eject
Publish
Build
Run on Release Configuration

**Comment by brentvatne** [comment]
hi! we need a reproducible example to help with this. please create a new issue and provide an example that we can run to demonstrate the issue. this is likely a bug in your own code and trying to provide a minimal reproducible example will help you to locate that bug! if you can reproduce it then it'll help us to investigate! more guidance on how to do this if you need it: https://stackoverflow.com/help/minimal-reproducible-example

**Comment by igaln** [comment]
it is a bit presumptuous for you to close this ticket just blaming our code. I gave details on how to replicate the issue, and there is a whole thread going on your forums of people having the same issue.

 If you had further questions we can give details and help others who are having similar issues.

The release 37 is ridden with bugs, but no blame peoples code...

**Comment by brentvatne** [comment]
@igaln - i am not blaming your code, i am just saying that we don't have enough information to be able to help you. if you can provide a reproducible example i would be happy to look into it

**Comment by brentvatne** [comment]
my guess at what is going on - the app.bundle and app.manifest files are at a different path than expected, so you can open up the ios project in xcode, remove the references, find them in your ios directory (in project-name/Supporting) and drag them back

**Comment by brentvatne** [comment]
published expo-cli@3.17.24 with a fix for my best guess at what might be going on in your situation, let me know if this resolves your issue. you can see a working example where i created an app, added fonts, then ejected and ran it in release mode on ios and android here https://github.com/brentvatne/ejected-example-fonts

**Comment by igaln** [comment]
will try it thank you. 

Just fyi, initially I was able to solve this issue by
removing the following key in the app.json
"assetBundlePatterns": [
  "**/*"
],

but this might be breaking the OTA updates I was told.

Will try the sample and the patch,
thanks again for all the input

**Comment by thierryskoda** [comment]
Just created a new expo app with expo-cli version 3.20.3, ejected, tried to run on a iOS simulator as a released build and I have the same error: 

`Terminating app due to uncaught exception ‘NSInvalidArgumentException’, reason: '* -[NSFileManager copyItemAtPath:toPath:options:error:]: source path is nil’**
`

I think expo-updates 2.0.0 is causing that error but I'm not sure.

**Comment by brentvatne** [comment]
@thierryskoda - can you please provide a reproducible example?

**Comment by thierryskoda** [comment]
Here's the repo https://github.com/usewalter/expo-eject-testing.

**Comment by brentvatne** [comment]
I cloned that project, ran `yarn`, ran `npx pod-install`, then opened up the xcworkspace and did a release build and it worked as expected. Might be worth running "Clean Build Folder" (cmd+shift+k) and rebuilding on your machine.

**Comment by thierryskoda** [comment]
Thanks @brentvatne!

@esamelson wrote a reply in this issue here https://github.com/expo/expo/issues/8271#issuecomment-628978265

Updated my cli, created a new brand new project, ejected it and did a release build and it worked.

Cleaning the build folder might worked also. Anyway... Now everything's working thank you!

**Comment by brentvatne** [comment]
awesome! glad you could figure it out! 💪

**Comment by thierryskoda** [comment]
Hey @brentvatne It's not working finally... :( 

Here's another repo I made to show the issue.
https://github.com/thierryskoda/testing-expo-2

Here's the step I did in this repo:

1. expo init
2. Chose the `tabs` template
3. Ejected
4. Opened the project in Xcode
5. Ran a release build on the simulator
6. The app crash with the same error as above.

From the tests I did, the error seem to come from `import { createStackNavigator } from '@react-navigation/stack'`

Whenever I remove that import, everything works fine.

**Comment by Firanus** [comment]
I'm experiencing a similar issue @brentvatne; after upgrading our app to expo updates 0.2.1, release builds on iOS crash on start up with the same `Terminating app due to uncaught exception ‘NSInvalidArgumentException’, reason: '* -[NSFileManager copyItemAtPath:toPath:options:error:]: source path is nil’**` reported above

**Comment by mdcuk34** [comment]
We're also having this issue and thought it was react-navigation so created an [issue](https://github.com/react-navigation/react-navigation/issues/8268) there. Everything above aligns with what we're experiencing. 

We’ve created a reproduction [here](https://github.com/maurovisintin/expo-bare-minimum) where the app will work out the box with a `yarn install`, `pod install` and built in release mode through Xcode. If you then uncomment the line import Navigation from './src/components/navigation'; and rebuild you will receive the above error.

**Comment by satya164** [comment]
React Navigation imports images, so try creating a project which also imports images without React Navigation and see if that causes issues.

**Comment by mdcuk34** [comment]
@satya164 Good shout, using an image in the project causes this error. (I'm on `expo-cli@3.20.7`).

@brentvatne Any ideas?

**Comment by esamelson** [comment]
hey everyone -- thanks for the clear repro case. we're looking into this and hope to have a fix out soon, will update here with more information!

**Comment by WoodburyShortridge** [comment]
Thanks @esamelson I'm hitting same issue:
`*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[NSFileManager copyItemAtPath:toPath:options:error:]: source path is nil'`

Importing images in bare workflow w/ updates crashes on launch.

**Comment by esamelson** [comment]
Hey all -- this issue has been fixed as of expo-updates@0.2.2, which has just been released. Thanks for the timely and helpful reports!
